encode.go

```diff
@@ -20,12 +20,31 @@ func (te *TypeError) Error() string {
 
 // An Encoder encodes Go data structures into pickle byte stream
 type Encoder struct {
-	w io.Writer
+	w      io.Writer
+	config *EncoderConfig
+}
+
+// EncoderConfig allows to tune Encoder.
+type EncoderConfig struct {
+	// PersistentID, if !nil, will be used by encoder to encode objects as persistent references.
+	//
+	// Whenever encoders sees pointer to a Go struct object, it will call
+	// PersistentID to find out how to encode that object. If PersistentID
+	// returns nil, the object is encoded regularly. If !nil - the object
+	// will be encoded as Ref with persistent ID set to what was returned.
+	//
+	// See Ref documentation for details and further pointers on the subject.
+	PersistentID func(obj interface{}) interface{}
```

Comment:
Would it make more sense to have this return `*Ref` instead? A `nil` value would still indicate to encode the object as normal.